java-squid/src/main/java/org/sonar/java/AbstractJavaClasspath.java

```diff
@@ -0,0 +1,192 @@
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+ */
+package org.sonar.java;
+
+import com.google.common.base.Splitter;
+import com.google.common.collect.Lists;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.io.IOCase;
+import org.apache.commons.io.filefilter.AndFileFilter;
+import org.apache.commons.io.filefilter.DirectoryFileFilter;
+import org.apache.commons.io.filefilter.FalseFileFilter;
+import org.apache.commons.io.filefilter.IOFileFilter;
+import org.apache.commons.io.filefilter.OrFileFilter;
+import org.apache.commons.io.filefilter.TrueFileFilter;
+import org.apache.commons.lang.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.sonar.api.BatchExtension;
+import org.sonar.api.batch.fs.FileSystem;
+import org.sonar.api.config.Settings;
+import org.sonar.api.resources.Project;
+import org.sonar.api.utils.WildcardPattern;
+
+import java.io.File;
+import java.io.FileFilter;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.List;
+
+import static org.apache.commons.io.filefilter.FileFilterUtils.suffixFileFilter;
+
+public abstract class AbstractJavaClasspath implements BatchExtension {
+
+  private static final char SEPARATOR = ',';
+  private static final Logger LOG = LoggerFactory.getLogger(AbstractJavaClasspath.class);
+  protected final Project project;
+  protected final Settings settings;
+  protected final FileSystem fs;
+
+  protected List<File> binaries;
+  protected List<File> elements;
+  protected boolean validateLibraries;
+  protected boolean hasJavaSources;
+  protected boolean initalized;
+
+  public AbstractJavaClasspath(Project project, Settings settings, FileSystem fs) {
+    this.project = project;
+    this.settings = settings;
+    this.fs = fs;
+    initalized = false;
+  }
+
+  protected abstract void init();
+
+  protected List<File> getFilesFromProperty(String property) {
+    List<File> result = Lists.newArrayList();
+    String fileList = settings.getString(property);
+    if (StringUtils.isNotEmpty(fileList)) {
+      Iterable<String> fileNames = Splitter.on(SEPARATOR).omitEmptyStrings().split(fileList);
+      for (String pathPattern : fileNames) {
+        boolean isLibraryProperty = property.endsWith("libraries");
+        List<File> libraryFilesForPattern = getFilesForPattern(fs.baseDir(), pathPattern, isLibraryProperty);
```

Comment:
As the `baseDir` won't change during loop execution, it may be more interesting to use a variable initialized before the "for" statement instead of accessing it every time.

Same remark for the `isLibraryProperty`.